src/cli/self_update/windows.rs

```diff
@@ -285,6 +282,75 @@ pub fn do_remove_from_path() -> Result<()> {
     _apply_new_path(new_path)
 }
 
+pub fn do_add_to_programs() -> Result<()> {
+    use std::path::PathBuf;
+
+    let key = RegKey::predef(HKEY_CURRENT_USER)
+        .create_subkey(r"Software\Microsoft\Windows\CurrentVersion\Uninstall\rustup")
+        .chain_err(|| ErrorKind::PermissionDenied)?
```

Comment:
Is that the only possible error that can occur?

Reply:
I didn't really think about it as I took it from the PATH handling functions, which also deals with the registry.

Looking through `src/cli/errors.rs` only `PermissionDenied` seems appropriate. Functions in winreg return an `io::Result`, which will also be displayed when an error occur.

Reply:
https://docs.rs/winreg/0.8.0/winreg/struct.RegKey.html#method.create_subkey is the docs, looking at the source its backed by https://docs.rs/winreg/0.8.0/src/winreg/lib.rs.html#387-389 which is backed by winapi_reg::RegCreateKeyExW which means we should look at https://docs.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regcreatekeyexw which helpfully is unclear.

Anyhow, I guess this will have to do for now, though I'm certain other errors can occur. shrug.

Reply:
An alternative would be to add a variant in cli::errors::ErrorKind for errors that occur when using the registry